docs/docs/guide/GA4-google-analytics.mdx

```diff
@@ -124,6 +130,115 @@ Within GrowthBook, navigate to the **_Analysis_** section, and then click on the
   style={{ width: 500, margin: "0 auto 10px auto" }}
 />
 
-Then add your BigQuery connection info. GrowthBook will pre-populate the SQL queries required to use your GA4 data. You can also add a custom SQL query if you want to use a different table or filter the data in some way as you like.
+Then add your BigQuery connection info. GrowthBook will pre-populate the SQL queries required to use your GA4 data. You
+can also add a custom SQL query if you want to use a different table or filter the data in some way as you like.
+
+:::note
+
+While GrowthBook will pre-populate the SQL queries for you, you may need to adjust the experiment query to match your
+data depending on the way you are tracking your experiments (see the **_trackingCallback_** below).
+
+:::
+
+Once connected, you'll be able to add any additional metrics or dimensions, and then you can use your GA4 data for your
+experiments. You can use all your existing events and tracking- GrowthBook only requires one additional tracking call
+when a user is exposed to an experiment.
+
+## Running experiments with GrowthBook and GA4
+
+With the data source connected, you can use the GrowthBook SDK run A/B tests. The SDK lets you run experiments where the
+variations are defined in code, and controlled from UI. Once implemented, the SDK will do the random assignments and send
+the experiment exposure event to GA4 based on the settings in the GrowthBook UI.
+
+:::info
+
+We do have a visual editor for creating experiments, but it is meant for simple experiments. Experiments that are more
+complex are best created with a change in the code.
+
+:::
+
+### SDK integration for GA4
+
+You can follow the guides on SDK integration for any of our supported languages [here](http://localhost:3200/lib/). The
+only special case for GA4 is getting the **user Id** and/or **Client ID** attribute, and adjusting the `trackingCallback` to
+send the experiment exposure event to GA4.
+
+#### Using GA's Client ID
+
+GA generates a **Client ID** (sometimes called a device Id) for each visitor, and then stores this unique ID in a cookie
+(ga\_). There are a few ways to grab this Id from your code. One of the easiest ways with Javascript is using the following code:
+
+```js
+const gaUserId = document.cookie.match(/_ga=(.+?);/)[1].split('.').slice(-2).join(".")
```

Comment:
FYI, I think you can also get it via:
```
  gtag('get', 'GA4_CLIENT_ID', function(clientId) {
    console.log('Client ID:', clientId);
  });
```
Maybe worth mentioning as I think this might be the preferred way (from GA's perspective)

Reply:
I tried that, but it's not working for me... 
<img width="416" alt="Screen Shot 2023-03-07 at 11 45 57 AM" src="https://user-images.githubusercontent.com/46107/223535800-7e9f2b50-dd29-44e2-a9db-53841ca6dd3b.png">